### DISCUSSION TITLE
Custom Fonts in Matplotlib

### DISCUSSION THREAD

**Discussion by boisgera** [pyodide/pyodide#2252] [Ideas]
Hi everyone,

First, let me say that I am very impressed of the extent to which Matplotlib "just works" with Pyodide (via https://starboard.gg/).
I'd like to know if there is a way to use custom fonts in Matplotlib. From what I understand, if https://github.com/pyodide/pyodide/issues/45 is the latest info on the subject, we are limited to those which are embedded by matplotlib ? AFAICT, that would be:
    
    'DejaVu Sans Mono' 
    'STIXNonUnicode' 
    'STIXNonUnicode' 
    'STIXSizeTwoSym' 
    'STIXSizeThreeSym' 
    'DejaVu Sans Mono' 
    'STIXSizeOneSym' 
    'cmb10' 
    'DejaVu Serif' 
    'cmr10' 
    'STIXSizeFourSym' 
    'STIXSizeFourSym' 
    'STIXNonUnicode' 
    'STIXGeneral' 
    'DejaVu Serif Display' 
    'DejaVu Sans Mono' 
    'DejaVu Sans' 
    'STIXGeneral' 
    'DejaVu Sans Mono' 
    'DejaVu Serif' 
    'STIXGeneral' 
    'DejaVu Serif' 
    'DejaVu Sans Display' 
    'DejaVu Sans' 
    'cmsy10' 
    'cmss10' 
    'DejaVu Serif' 
    'cmtt10' 
    'cmmi10' 
    'cmex10' 
    'STIXSizeFiveSym' 
    'STIXSizeTwoSym' 
    'DejaVu Sans' 
    'STIXNonUnicode' 
    'STIXSizeOneSym' 
    'DejaVu Sans' 
    'STIXSizeThreeSym' 
    'STIXGeneral' 
    'Humor Sans' 

The ability to add fonts at runtime would be very nice 😊

Cheers !

Sébastien

**Comment by JosephBARBIERDARNAL** [comment]
Hi there,

I'm also interested in content on how to change matplotlib fonts with Pyodide.

**Reply by hoodmane** [reply]
Unfortunately, nobody really knows. Pyodide's matplotlib support is currently unmaintained:
https://github.com/pyodide/matplotlib-pyodide/issues/65

**Reply by JosephBARBIERDARNAL** [reply]
Thanks for mentioning it! 

FYI, I created a package to work more easily with fonts in matplotlib called [pyfonts](https://github.com/JosephBARBIERDARNAL/pyfonts) (IMO working with fonts has never been easy in matplotlib, even outside of pyodide). But I haven't managed to get it to work in pyodide yet.
